lntest/itest/lnd_test.go

```diff
@@ -10330,19 +10330,77 @@ func subscribeGraphNotifications(t *harnessTest, ctxb context.Context,
 	}
 }
 
+func waitForGraphSync(t *harnessTest, node *lntest.HarnessNode) {
+	err := wait.Predicate(func() bool {
+		ctxb := context.Background()
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		info, err := node.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+		require.NoError(t.t, err)
+
+		return info.SyncedToGraph
+	}, 10*time.Second)
+	require.NoError(t.t, err)
+}
+
 func testGraphTopologyNotifications(net *lntest.NetworkHarness, t *harnessTest) {
+	t.t.Run("pinned", func(t *testing.T) {
+		ht := newHarnessTest(t, net)
+		testGraphTopologyNtfns(net, ht, true)
+	})
+	t.t.Run("unpinned", func(t *testing.T) {
+		ht := newHarnessTest(t, net)
+		testGraphTopologyNtfns(net, ht, false)
+	})
+}
+
+func testGraphTopologyNtfns(net *lntest.NetworkHarness, t *harnessTest, pinned bool) {
 	ctxb := context.Background()
 
 	const chanAmt = funding.MaxBtcFundingAmount
 
-	alice, err := net.NewNode("alice", nil)
-	require.NoError(t.t, err)
-	defer shutdownAndAssert(net, t, alice)
-
+	// Spin up Bob first, since we will need to grab his pubkey when
+	// starting Alice to test pinned syncing.
 	bob, err := net.NewNode("bob", nil)
 	require.NoError(t.t, err)
 	defer shutdownAndAssert(net, t, bob)
 
+	// For unpinned syncing, start Alice as usual. Otherwise grab Bob's
+	// pubkey to include in his pinned syncer set.
+	var aliceArgs []string
+	if pinned {
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		bobInfo, err := bob.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+		require.NoError(t.t, err)
+
+		aliceArgs = []string{
+			"--numgraphsyncpeers=1",
+			fmt.Sprintf("--gossip.pinned-syncers=%s",
+				bobInfo.IdentityPubkey),
+		}
+	}
+
+	alice, err := net.NewNode("alice", aliceArgs)
+	require.NoError(t.t, err)
+	defer shutdownAndAssert(net, t, alice)
+
+	// When testing pinnned syncing, we need to ensure Alice doesn't select
```

Comment:
I don't get this. If Bob is the pinned peer, why do we need to check that Bob won't also be the active peer? Aren't we checking that Alice is syncing from dummy below, while we should be checking she syncs from Bob?

Reply:
Alice will complete an initial sync with the dummy, which means her graph will also be empty (but `synced_to_graph` will show true).

This is mainly done to avoid false positives due to regressions, since the test will pass all the same if we are unintentionally in ACTIVE_SYNC with Bob.

Reply:
opted for more explicit check of the sync state, ptal